README.md

```diff
@@ -275,6 +283,58 @@ connector.connect(
 )
 ``` 
 
+### Async Driver Usage
+The Cloud SQL Connector for Python currently supports the
+[asyncpg](https://magicstack.github.io/asyncpg) Postgres database driver.
+This driver leverages [asyncio](https://docs.python.org/3/library/asyncio.html)
+to improve the speed and efficiency of database connections through concurrency.
+
+The Cloud SQL Connector has an async `create_async_connector` that can be used
+and is recommended for async drivers as it returns a `Connector` object
+that uses the current thread's running event loop automatically.
+
+The `create_async_connector` allows all the same input arguments as the
+[Connector](#configuring-the-connector) object.
+
+Once a `Connector` object is returned by `create_async_connector` you can call
+its `connect_async` method, just as you would the `connect` method:
+
+```python
+import asyncpg
+from google.cloud.sql.connector import create_async_connector
+
+async def main():
+    # intialize Connector object using 'create_async_connector'
+    connector = await create_async_connector()
+
+    # create connection to Cloud SQL database
+    conn: asyncpg.Connection = await connector.connect_async(
```

Comment:
nit: this connection should probably use an async with

Reply:
the connector or connection itself? `async with create_async_connector()` or `async with connector.connect_async(...)` ?

Reply:
The connection itself. 

Do we support async enters and exit with the Connector?

Reply:
not currently, we don't have an `__aenter__` or `__aexit__` implemented.

Reply:
I added the ability to create an async context manager with `Connector` and added example to README. I can't use `async with` for the connection because the [asyncpg.Connection](https://github.com/MagicStack/asyncpg/blob/master/asyncpg/connection.py#L42) class does not have an `__aenter__` or `__aexit__`. We could potentially create a `Connection` wrapper class of our own to implement an async context manager for the database connection. ([Example](https://stackoverflow.com/questions/62507653/python-context-manager-for-asyncpg)). However, if in a future PR we are going to support connection pools (either through SQLAlchemy or native `asyncpg.Pool` then they will have context manager built in so not sure it is worth it to create the wrapper class.

Reply:
Ok, this is surprising to me that they don't support those two functions. I think it's fine to leave for now (maybe worth asking why they haven't?)

Reply:
Seems like there is an [open issue](https://github.com/MagicStack/asyncpg/issues/760) for it actually, if I have some spare time I will try and put up another PR to asyncpg and add the two functions.